backend/crates/eiffelvis_core/src/types.rs

```diff
@@ -0,0 +1,43 @@
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+pub struct BaseData {}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+pub struct BaseMeta {
+    pub id: Uuid,
+    #[serde(rename = "type")]
+    pub event_type: String,
+    pub version: String,
+    pub time: u64,
+}
+
+#[derive(Default, Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Serialize, Deserialize)]
+pub struct BaseLink {
+    #[serde(rename = "type")]
+    pub link_type: String,
+    pub target: Uuid,
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+pub struct BaseEvent {
```

Comment:
Having an eiffelvis_types crate seemed a bit silly so collapsed it into eiffelvis_core considering anyone needing these types likely also depends on core